cmd/traffic/cmd/manager/service.go

```diff
@@ -573,60 +576,65 @@ func (m *Manager) GetLogs(ctx context.Context, request *rpc.GetLogsRequest) (*rp
 		PodLogs: make(map[string]string),
 	}
 	var errMsg string
-	client := managerutil.GetKatesClient(ctx)
+	clientset := managerutil.GetK8sClientset(ctx)
 
 	// getPodLogs is a helper function for getting the logs from the container
 	// of a given pod. If we are unable to get a log for a given pod, we will
 	// instead return the error in the map, instead of the log, so that:
 	// - one failure doesn't prevent us from getting logs from other pods
 	// - it is easy to figure out why gettings logs for a given pod failed
-	getPodLogs := func(pods []*kates.Pod, container string) {
+	getPodLogs := func(pods []*corev1.Pod, container string) {
+		wg := sync.WaitGroup{}
+		wg.Add(len(pods))
 		for _, pod := range pods {
-			logEvents := make(chan kates.LogEvent)
-			plo := &kates.PodLogOptions{
-				Container: container,
-			}
-			// Since the same named workload could exist in multiple namespaces
-			// we add the namespace into the name so that it's easier to make
-			// sense of the logs
-			podAndNs := fmt.Sprintf("%s.%s", pod.Name, pod.Namespace)
-			if err := client.PodLogs(ctx, pod, plo, false, logEvents); err != nil {
-				resp.PodLogs[podAndNs] = fmt.Sprintf("Failed to get logs: %s", err)
-				continue
-			}
-			var log string
-			for {
-				event := <-logEvents
-				// If there's an error, we want to put that in the log output for that
-				// container.
-				if event.Error != nil {
-					log += fmt.Sprintf("Failed getting log event: %s\n", event.Error)
+			go func(pod *corev1.Pod) {
+				dlog.Errorf(ctx, "Getting logs for container: %s in pod: %s", container, pod.Name)
```

Comment:
Is this Errorf so that it itself shows up in the log collection for the traffic manager? It seems like a weird log level otherwise

Reply:
(no change needed here btw, just curious 🙂 )

Reply:
Oh whoops that was something I added when I was debugging the switch over the client-go, going to remove